diff -ru ghdl-1.0.0/configure ghdl-1.0.0_multilib/configure
--- ghdl-1.0.0/configure	2021-02-02 20:52:50.000000000 +0100
+++ ghdl-1.0.0_multilib/configure	2021-03-29 13:03:47.748029391 +0200
@@ -33,6 +33,7 @@
 prefix=/usr/local
 libdirsuffix=lib/ghdl
 libdirreverse=../..
+libdirplatform=lib
 gcc_src_dir=
 llvm_config=
 backtrace_lib=
@@ -52,7 +53,7 @@
 show_help=no
 progname=$0
 
-subst_vars="CC CXX GNATMAKE MAKE CFLAGS LDFLAGS build srcdir abs_srcdir prefix backend libdirsuffix libdirreverse gcc_src_dir llvm_config llvm_be backtrace_lib build_mode EXEEXT SOEXT PIC_FLAGS default_pic enable_werror enable_checks enable_gplcompat enable_libghdl libghdl_version ghdl_version"
+subst_vars="CC CXX GNATMAKE MAKE CFLAGS LDFLAGS build srcdir abs_srcdir prefix backend libdirsuffix libdirreverse libdirplatform gcc_src_dir llvm_config llvm_be backtrace_lib build_mode EXEEXT SOEXT PIC_FLAGS default_pic enable_werror enable_checks enable_gplcompat enable_libghdl libghdl_version ghdl_version"
 
 # Find srcdir
 srcdir=`dirname $progname`
@@ -221,6 +222,11 @@
 fi
 echo "Build machine is: $build"
 
+case "$build" in
+    *64*) libdirplatform=lib64 ;;
+    *) libdirplatform=lib ;;
+esac
+
 # For mcode, check that gcc emits i386
 if test $backend = mcode; then
   case "$build" in
diff -ru ghdl-1.0.0/Makefile.in ghdl-1.0.0_multilib/Makefile.in
--- ghdl-1.0.0/Makefile.in	2021-02-02 20:52:50.000000000 +0100
+++ ghdl-1.0.0_multilib/Makefile.in	2021-03-29 13:04:26.606030501 +0200
@@ -24,6 +24,7 @@
 backend=@backend@
 libdirsuffix=@libdirsuffix@
 libdirreverse=@libdirreverse@
+libdirplatform=@libdirplatform@
 EXEEXT=@EXEEXT@
 SOEXT=@SOEXT@
 PIC_FLAGS=@PIC_FLAGS@
@@ -46,7 +47,7 @@
 PWD?=$(shell pwd)
 DESTDIR=
 bindir=$(prefix)/bin
-libdir=$(prefix)/lib
+libdir=$(prefix)/$(libdirplatform)
 incdir=$(prefix)/include
 MKDIR=mkdir
 LN=ln -s
