Place .so files in lib64 for x86_64 systems.

diff -ru ghdl-0.37/configure ghdl-0.37_multilib/configure
--- ghdl-0.37/configure	2020-02-28 22:25:18.000000000 +0100
+++ ghdl-0.37_multilib/configure	2020-10-27 23:52:15.182101607 +0100
@@ -35,6 +35,7 @@
 prefix=/usr/local
 libdirsuffix=lib/ghdl
 libdirreverse=../..
+libdirplatform=lib
 gcc_src_dir=
 llvm_config=
 backtrace_lib=
@@ -54,7 +55,7 @@
 show_help=no
 progname=$0
 
-subst_vars="CC CXX GNATMAKE MAKE CFLAGS LDFLAGS build srcdir abs_srcdir prefix backend libdirsuffix libdirreverse gcc_src_dir llvm_config llvm_be backtrace_lib build_mode EXEEXT SOEXT PIC_FLAGS default_pic enable_werror enable_checks enable_openieee enable_libghdl libghdl_version ghdl_version"
+subst_vars="CC CXX GNATMAKE MAKE CFLAGS LDFLAGS build srcdir abs_srcdir prefix backend libdirsuffix libdirreverse libdirplatform gcc_src_dir llvm_config llvm_be backtrace_lib build_mode EXEEXT SOEXT PIC_FLAGS default_pic enable_werror enable_checks enable_openieee enable_libghdl libghdl_version ghdl_version"
 
 # Find srcdir
 srcdir=`dirname $progname`
@@ -223,6 +224,11 @@
 fi
 echo "Build machine is: $build"
 
+case "$build" in
+    *64*) libdirplatform=lib64 ;;
+    *) libdirplatform=lib ;;
+esac
+
 # For mcode, check that gcc emits i386
 if test $backend = mcode; then
   gcc_machine=`$CC $CFLAGS -dumpmachine`
diff -ru ghdl-0.37/Makefile.in ghdl-0.37_multilib/Makefile.in
--- ghdl-0.37/Makefile.in	2020-02-28 22:25:18.000000000 +0100
+++ ghdl-0.37_multilib/Makefile.in	2020-10-27 23:52:15.182101607 +0100
@@ -26,6 +26,7 @@
 backend=@backend@
 libdirsuffix=@libdirsuffix@
 libdirreverse=@libdirreverse@
+libdirplatform=@libdirplatform@
 EXEEXT=@EXEEXT@
 SOEXT=@SOEXT@
 PIC_FLAGS=@PIC_FLAGS@
@@ -48,7 +49,7 @@
 PWD?=$(shell pwd)
 DESTDIR=
 bindir=$(prefix)/bin
-libdir=$(prefix)/lib
+libdir=$(prefix)/$(libdirplatform)
 incdir=$(prefix)/include
 MKDIR=mkdir
 LN=ln -s
